nx_parallel/algorithms/shortest_paths/dense.py

```diff
@@ -0,0 +1,251 @@
+                if is_prime and block == no_of_primary - 1:
+                    block_coord = (block * blocking_factor, n - 1)
+                else:
+                    block_coord = _block_range(blocking_factor, block)
+                params.append((block_coord, k))
+                params.append((k, block_coord))
+
+        Parallel(n_jobs=(no_of_primary - 1) * 2, require="sharedmem")(
+            delayed(_partial_floyd_warshall_numpy)(A, k, i, j) for (i, j) in params
+        )
+
+        # Phase 3: Compute remaining
+        params.clear()
+        for block_i in range(no_of_primary):
+            for block_j in range(no_of_primary):
+                # skip all block previously computed, so skip every block with primary block value
+                if block_i != primary_block or block_j != primary_block:
+                    i_range = _block_range(blocking_factor, block_i)
+                    j_range = _block_range(blocking_factor, block_j)
+                    if is_prime:
+                        if block_i == no_of_primary - 1:
+                            i_range = (block_i * blocking_factor, n - 1)
+                        if block_j == no_of_primary - 1:
+                            j_range = (block_j * blocking_factor, n - 1)
+                    params.append((i_range, j_range))
+        Parallel(n_jobs=(no_of_primary - 1) ** 2, require="sharedmem")(
+            delayed(_partial_floyd_warshall_numpy)(A, k, i, j) for (i, j) in params
+        )
+    dist = _matrix_to_dict(A, nodelist)
+
+    return dist
+
+
+def _partial_floyd_warshall_numpy(A, k_iteration, i_iteration, j_iteration):
+    """
+    Compute partial FW in the determined sub-block for the execution of
+    parallel tiled FW.
+
+    Parameters
+    ----------
+    A : 2D numpy.ndarray
+        matrix that reppresent the adjacency matrix of the graph
+
+    k_iteration : tuple
+        range (start-end) of the primary block in the current iteration
+
+    i_iteration : tuple
+        range (start-end) of the rows in the current block computed
+
+    j_iteration : tuple
+        range (start-end) of the columns in the current block computed
+
+    Returns
+    -------
+    A : 2D numpy.ndarray
+        adjacency matrix updated after floyd warshall
+    """
+    for k in range(k_iteration[0], k_iteration[1] + 1):
+        for i in range(i_iteration[0], i_iteration[1] + 1):
+            for j in range(j_iteration[0], j_iteration[1] + 1):
+                A[i][j] = min(A[i][j], (A[i][k] + A[k][j]))
+
+
+def _block_range(blocking_factor, block):
+    return (block * blocking_factor, (block + 1) * blocking_factor - 1)
+
+
+def _calculate_divisor(i, x, y):
+    if x % i == 0:
+        divisor1 = result2 = i
+        result1 = divisor2 = x // i
+        # difference1 = abs((result1 - 1) ** 2 - y)
+        difference1 = abs(result1 - y)
+        # difference2 = abs((result2 - 1) ** 2 - y)
+        difference2 = abs(result2 - y)
+
+        if difference1 < difference2:
```

Comment:
Could be missing something obvious, but aren't `divisor1` and `divisor2` actually going to be the same thing here (`i`)? i.e.  do we even need to compare difference1 with difference2?

I wonder whether this could be simplified to something like:
```
def _calculate_divisor(i, x, y):
    if x % i == 0:
        quotient = x // i
        return (i, quotient, abs(i - y))
    return None
```
?

Reply:
Not really the same thing, the idea is to find the divisor of x ( size of matrix) most similar to y (core available).
To find i try to divide x to all number from 1 to sqrt(x) so when i do x // i, divisor1 is gonna be i, the result of x // i is gonna be divisor2.

An example to clarify:
x=10
y=4
i=2
 x // i = 5
So divisor1 = 2, divisor2 = 5

divisor2 is the most similar to 4 so i choose 5 as divisor

Reply:
Thanks for the clarification on this @Fre0Grella 

I think what I'm still struggling with is the need for two separate comparisons. Since `divisor1` is `i` and `divisor2` is `x // i`, aren't you always comparing `i` and `x // i`? The current implementation is functionally correct, but this seems like it could be greatly simplified without losing the intent of matching `divisor2` more closely to `y`.

For example, maybe we could directly compute the abs difference of both `i` and `x // i` with `y`, then just return whichever is closest? And this would avoid the extra complexity in checking two sets of variables. So, building on my (incomplete) alternative suggestion previously, this might look something like:
```
def _calculate_divisor(i, x, y):
    if x % i == 0:
        divisor = x // i
        return (i, divisor, min(abs(i - y), abs(divisor - y)))
    return None
```
keeps the logic intact but then we don't need to assign and compare `divisor1` and `divisor2` explicitly. Let me know what you think / if I'm missing something!